lib/extensions.js

```diff
@@ -104,6 +104,19 @@ module.exports = function(realmConstructor) {
 
         setConstructorOnPrototype(realmConstructor.Sync.User);
         setConstructorOnPrototype(realmConstructor.Sync.Session);
+
+        if (realmConstructor.Sync._setFeatureToken) {
+            realmConstructor.Sync.setFeatureToken = function(featureToken) {
+                if (typeof featureToken !== 'string' || !featureToken instanceof String) {
+                    throw new Error("featureToken should be a string");
+                }
+
+                realmConstructor.Sync._setFeatureToken(featureToken.trim());
+            }
+
+            //enable depreciated setAccessToken
```

Comment:
done. 
clearly I don't know how to write since my branches are named like that as well :)